runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkBatchTransformTranslators.java

```diff
@@ -466,36 +495,55 @@ private static void rejectSplittable(DoFn<?, ?> doFn) {
 
   private static class ParDoTranslatorBatch<InputT, OutputT>
       implements FlinkBatchPipelineTranslator.BatchTransformTranslator<
-      ParDo.MultiOutput<InputT, OutputT>> {
+      PTransform<PCollection<InputT>, PCollectionTuple>> {
 
     @Override
     @SuppressWarnings("unchecked")
     public void translateNode(
-        ParDo.MultiOutput<InputT, OutputT> transform,
+        PTransform<PCollection<InputT>, PCollectionTuple> transform,
         FlinkBatchTranslationContext context) {
-      DoFn<InputT, OutputT> doFn = transform.getFn();
+      DoFn<InputT, OutputT> doFn;
+      try {
+        doFn = (DoFn<InputT, OutputT>) ParDoTranslation.getDoFn(context.getCurrentTransform());
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
       rejectSplittable(doFn);
       DataSet<WindowedValue<InputT>> inputDataSet =
           context.getInputDataSet(context.getInput(transform));
 
       Map<TupleTag<?>, PValue> outputs = context.getOutputs(transform);
 
+      TupleTag<?> mainOutputTag;
+      try {
```

Comment:
Kind of cluttery having this construct everywhere. I wonder if there is a way to lift the exception catching.

Reply:
Is java 8 supported by all the submodules? The only clean way I can think of is to demote the exception to runtime using a lambda wrapper.

Reply:
I've gone the lambda route to address this. It's fairly general and will be needed again in the streaming translator. However, I've left it inside of FlinkBatchTransformTranslators for now because it's unclear to me whether Java 8 is allowed globally.

Please advise on a better location/name.

Reply:
Looks like I wrote that too soon. The Maven build appears to be configured to use `-source 1.7`, at least for this module. I'm not sure what else to do.